### ISSUE TITLE
slack logger

### ISSUE THREAD

**Issue by steenzout** [mediaFORGE/gol#11]
Slack logger implementation.

**Comment by david4shure** [comment]
Is the idea behind this to be able to asynchronously log to slack?

**Comment by steenzout** [comment]
Yes. Interested in this?
On Jan 28, 2016 9:59 AM, David Shure <notifications@github.com> wrote:Is the idea behind this to be able to asynchronously log to slack?

**Comment by david4shure** [comment]
Yeah! So would the use case for these different output adapters be that somebody wants to monitor a process in a slack channel (kinda similar to how jenkins handles output)? Seems like that would be a cool integration.

**Comment by david4shure** [comment]
Seems like we can make an abstraction layer between these adapters so that we can eliminate code rewriting as well. Also I found a go slack api here

https://github.com/nlopes/slack

**Comment by david4shure** [comment]
We would need to verify that is safe to use with go channels however.

**Comment by steenzout** [comment]
@david4shure the objective was to produce a library with several interfaces so you can easily perform asynchronous logging with the ability to easily format your messages and send them to anything you want (syslog, file, socket, chat service, etc).

For this issue in particular I believe we would only need to implement the `Logger` interface just like in [here](https://github.com/mediaFORGE/gol/blob/develop/loggers/syslog/syslog.go).

Since `Logger` is an interface you can make a `SlackLogger` and host it on your own repo using any 3rd party library you want or contribute one to this repo.
